sklearn/preprocessing/data.py

```diff
@@ -3147,3 +3150,216 @@ def inverse_transform(self, X):
                     X_tr[mask, idx] = None
 
         return X_tr
+
+
+class UnaryEncoder(BaseEstimator, TransformerMixin):
+    """Encode ordinal integer features using a unary scheme.
+
+    The input to this transformer should be a matrix of non-negative integers,
+    denoting the values taken on by ordinal (discrete) features. The output
+    will be a matrix where each column corresponds to one possible value of
+    one feature. It is assumed that input features take on values in the range
+    0 to (n_values - 1).
+
+    This encoding is needed for feeding ordinal features to many scikit-learn
+    estimators, notably linear models and kernel-based models like SVMs with
+    the standard kernels.
+    This transformation is unlikely to help when using with tree-based models,
+    since those already work on the basis of a particular feature value being
+    < or > than a threshold, unlike linear and kernel-based models.
+
+    Read more in the :ref:`User Guide <preprocessing_ordinal_features>`.
+
+    Parameters
+    ----------
+    n_values : 'auto', int or array of ints
+        Number of values per feature.
+
+        - 'auto' : determine value range from training data.
+        - int : number of ordinal values per feature.
+                Each feature value should be in ``range(n_values)``
+        - array : ``n_values[i]`` is the number of ordinal values in
+                  ``X[:, i]``. Each feature value should be
+                  in ``range(n_values[i])``
+
+    ordinal_features : "all" or array of indices or mask
+        Specify what features are treated as ordinal.
+
+        - 'all' (default): All features are treated as ordinal.
+        - array of indices: Array of ordinal feature indices.
+        - mask: Array of length n_features and with dtype=bool.
+
+        Non-ordinal features are always stacked to the right of the matrix.
+
+    dtype : number type, default=np.float
+        Desired dtype of output.
+
+    sparse : boolean, default=False
+        Will return sparse matrix if set True else will return an array.
+
+    handle_greater : str, 'warn' or 'error' or 'clip'
+        Whether to raise an error or clip or warn if an
+        ordinal feature >= n_values is passed in.
+
+        - 'warn' (default): same as clip but with warning.
```

Comment:
For CategoricalEncoder the default is `error` (and for OneHotEncoder as well). Is there a good reason to deviate in this case?

Reply:
I see the discussion about this, @jnotham saying:

> I now wonder if we should have a default handle_greater='warn'. I think in practice that raising an error when a count-valued feature exceeds its training set range is too intrusive. Better off clipping but warning unless the user has specified otherwise.

I understand that reasoning for count-like values. But the examples in the documentation are not count-like, and for those this makes less sense I think as a default. 
(but don't know with which type of data this will be used more often)